promote.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/urfave/cli"
+)
+
+// PromoteApplication moves an application from one environment to another
+func PromoteApplication(c *cli.Context) error {
+	type environment struct {
+		Name string `json:"name"`
+		UUID string `json:"uuid"`
+	}
+
+	type appDetails struct {
+		Name         string        `json:"name"`
+		UUID         string        `json:"uuid"`
+		Environments []environment `json:"environments"`
+	}
+
+	appName := c.Args().Get(0)
+	envSrcName := c.Args().Get(1)
+	envDstName := c.Args().Get(2)
+	var envSrc environment
+	var envDst environment
```

Comment:
We don't need to initialize these yet; we may not even use them if we return because of an error.